src/mmrelay/config.py

```
@@ -274,172 +203,227 @@ def _add(path: str | None) -> None:
+    If the unified store resolver signals the platform does not support E2EE, this function returns a legacy store path under the user's home directory and logs a warning. If directory creation fails due to filesystem permissions or other OS errors, it returns the same home-based fallback path and logs a warning instead of failing. If the unified resolver raises a RuntimeError for reasons other than platform support, that RuntimeError is propagated.
+
+    Returns:
+        The absolute path to the E2EE store directory as a string.
+
+    Raises:
+        RuntimeError: If the unified store resolver raises a RuntimeError that should not be handled as a platform-unsupported condition.
+    """
+    try:
+        store_dir = str(get_unified_store_dir())
+        os.makedirs(store_dir, exist_ok=True)
+        return store_dir
```

Ruff (0.14.14)

[warning] Consider moving this statement to an `else` block

(TRY300)